drivers/ethernet/phy/phy_mii.c

```diff
@@ -467,7 +474,10 @@ static int phy_mii_initialize(const struct device *dev)
 
 		data->state.speed = speed_to_phy_link_speed[cfg->fixed_speed];
 		data->state.is_up = true;
+#if NOT_ALL_FIXED_LINK
```

Comment:
the whole code in this block should be a separate function now,  I do not think giant maybe-existing else blocks are a good paradigm, things like this will make it harder for people to contribute later , and when they do they will 90% of the time be adding to the mess, so as maintainer we have to take the opportunity to clean it up every time we make any change when possible

Reply:
I separated both modes into their own init functions